### PR TITLE
Adding the report mail and slack notifications

### PR DESCRIPTION
**Pull request by agutierrez-eb**
In this PR, we added the email report and slack notifications to generation process (with new architecture), with tests. Also, we added the email report to the declaration script. 
We've changed the patchs in the new architecture's tests too.

### CODE REVIEW

File: invoicing_app/mail_report_module.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class DeclarationProccessMailReport():
+
+    def declaration_send_email_report(self, start_date_period, end_date_period, currency):
+        start_date = start_date_period - relativedelta(days=1)
+        end_date = end_date_period - relativedelta(seconds=1)
+
+        report_data = TaxReceipt.objects.filter(
+            currency=currency,
+            start_date_period__range=(start_date, end_date),
+            status_id=3,
```
Comment (fgomez-eb) [inline invoicing_app/mail_report_module.py:53]: You have to replace the 3 for this TaxReceiptStatuses.get_id_from_name("PROCESSED")

File: invoicing_app/templates/declaration_template.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<html>
+	<body>
+		<h1>Declared Tax Receipts</h1>
+		<p>Here is the report from period the {{ period }} :</p>
+		<p>Country: {{ country }}</p>
+        <p>DECLARED GTF: {{ declared_gtf }}</p>
+		<p>Thanks!</p>
+	</body>
+</html>
```
Comment (fgomez-eb) [inline invoicing_app/templates/declaration_template.html]: here you have to add a space at the end of the file

File: invoicing_app/templates/generation_template.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	<body>
+		<h1>Generate Tax Recepits</h1>
+		<p>Here is the report from period the {{ period }} :</p>
+		<p>Country: {{ country }}</p>
+		<p>Number of events: {{ count_id }}</p>
+        <p>GTS: {{ gts }}</p>
+        <p>GTF: {{ gtf }}</p>
+		<p>Thanks!</p>
+	</body>
+</html>
```
Comment (fgomez-eb) [inline invoicing_app/templates/generation_template.html]: ditto

**Review by fgomez-eb — CHANGES_REQUESTED** [review]

File: invoicing_app/templates/declaration_template.html (hunk 2 above)
Comment (agutierrez-eb) [reply invoicing_app/templates/declaration_template.html]: Done!

File: invoicing_app/mail_report_module.py (hunk 1 above)
Comment (agutierrez-eb) [reply invoicing_app/mail_report_module.py:53]: I think in our local we don't have the TaxReceiptStatuses model. When we go to prod, we will have to check this changes, but for now i local, i think it's ok with this status.

**Review by ghidalgo-eb — APPROVED** [review]

**Review by fgomez-eb — APPROVED** [review]

**Pull request merged**